src/db/tables/onerep_scans.ts

```diff
@@ -69,84 +97,45 @@ async function setOnerepManualScan(
 
 async function addOnerepScanResults(
   onerepProfileId: number,
-  onerepScanId: number,
   onerepScanResults: Array<ScanResult>,
-  onerepScanReason: Scan["reason"],
-  onerepScanStatus: Scan["status"],
 ) {
-  await knex.transaction(async (transaction) => {
-    if (onerepScanReason === "manual") {
-      // Manual scans update an existing scan, replacing the previous results:
-      logger.info("manual_scan_updated", {
-        onerepScanId,
-        onerepScanReason,
-        onerepScanStatus,
-      });
-      await transaction("onerep_scan_results")
-        .delete()
-        .where("onerep_scan_id", onerepScanId);
-    }
-
-    // Create a new scan if it does not already exist. If it already exists:
-    // Update the status of the scan.
-    logger.info("scan_created_or_updated", {
-      onerepScanId,
-      onerepScanReason,
-      onerepScanStatus,
-    });
-    await transaction("onerep_scans")
-      .insert({
-        onerep_profile_id: onerepProfileId,
-        onerep_scan_id: onerepScanId,
-        onerep_scan_reason: onerepScanReason,
-        onerep_scan_status: onerepScanStatus,
-        // @ts-ignore knex.fn.now() results in it being set to a date,
-        // even if it's not typed as a JS date object:
-        created_at: knex.fn.now(),
-      })
-      .onConflict("onerep_scan_id")
-      .merge({
-        onerep_scan_status: onerepScanStatus,
```

Comment:
When a scan is finished and we are setting the scan results we should still update the the latest scan status. Otherwise, we are reverting the change introduced by https://github.com/mozilla/blurts-server/pull/3470 — which we rely on for showing the “scan in progress” dashboard state.

Reply:
Thanks - will add that rn.

Reply:
OK I did this in a different place, `setOnerepScan`, which I think is clearer than doing it as part of `addOnerepScanResults` ... now we're back in the "set" vs. "add" sort of discussion :) I think we should revisit soon whether we want to just make this an append operation.